src/js/pages/jobs/JobsTaskDetail.js

```diff
@@ -0,0 +1,178 @@
+import TaskDetail from '../services/task-details/TaskDetail';
+import TaskStates from '../../constants/TaskStates';
+
+/* eslint-disable no-unused-vars */
+const METHODS_TO_BIND = [
+  'handleKillButtonClick'
+];
+/* eslint-enable no-unused-vars */
+
+const JOBS_TABS = {
+  'jobs-task-details-tab': 'Details',
+  'jobs-task-details-files': 'Files',
+  'jobs-task-details-logs': 'Logs'
+};
+
+const MORE_METHODS_TO_BIND = [
+  'handleKillButtonClick',
+  'handleKillModalClose',
+  'handleTaskKill'
+]
+
+class JobsTaskDetail extends TaskDetail {
+  constructor() {
+    super(...arguments);
+
+    this.store_listeners = [
+      {name: 'state', events: ['success'], listenAlways: false},
+      {name: 'summary', events: ['success'], listenAlways: false},
+      {name: 'taskDirectory', events: ['error', 'success']},
+      {name: 'chronos', events: ['jobSuspendSuccess', 'jobSuspendError']}
+    ];
+
+    this.state = {
+      directory: null,
+      selectedLogFile: null,
+      errorCount: 0,
+      openKillConfirm: false,
+      killError: null
+    };
+
+    MORE_METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    super.componentWillMount(...arguments);
+
+    this.tabs_tabs = Object.assign({}, JOBS_TABS);
+    this.updateCurrentTab();
+  }
+
+  onChronosStoreJobSuspendError() {
+    this.setState({killError: true});
+  }
+
+  onChronosStoreJobSuspendSuccess() {
+    this.handleKillModalClose();
+  }
+
+  handleKillButtonClick() {
+    this.setState({openKillConfirm: true});
+  }
+
+  handleKillModalClose() {
+    this.setState({
+      openKillConfirm: false,
+      killError: false
+    });
+  }
+
+  handleTaskKill() {
+    ChronosStore.suspendSchedule(this.props.params.taskID);
+  }
+
+  getConfirmModal() {
+    return (
+      <Confirm
+        closeByBackdropClick={true}
+        footerContainerClass="container container-pod container-pod-short
+          container-pod-fluid flush-top flush-bottom"
+        open={this.state.openKillConfirm}
+        onClose={this.handleKillModalClose}
+        leftButtonCallback={this.handleKillModalClose}
+        leftButtonText="Cancel"
+        rightButtonCallback={this.handleTaskKill}
+        rightButtonClassName="button button-danger"
+        rightButtonText="Kill">
+        {this.getConfirmModalContent()}
+      </Confirm>
+    )
+  }
+
+  getConfirmModalContent() {
+    let error = '';
+
+    if (this.state.killError) {
+      error = (
+        <p className="text-danger text-small text-align-center">
+          An error has occurred.
+        </p>
+      );
+    }
+
+    return (
+      <div className="container-pod container-pod-short text-align-center">
+        <h3 className="flush-top">Are you sure?</h3>
+        {error}
+        <p>
+          {`${this.props.params.taskID} will be killed.`}
+        </p>
+      </div>
+    );
+  }
+
+  getJob() {
+    return ChronosStore.getJob(this.props.params.id);
+  }
```

Comment:
I don't think you need to have a separate function for this. You can just have this one line in the method below since it's only used once.